[mogmyij] iP

mogmyij opened the pull request:
# Mab Chatbot 

> _a successor to Duke_ 🥇 ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ))

## What can ~~Duke~~ Mab do?
- Save tasks to do
- Save tasks with deadlines
- Save events with beginning and end timing
- persistant data storage

## For people who are:

1. forgetful
2. lazy
3. anxious about forgetting tasks

87160510 Features
- [x] Task saving
- [ ] Task reminders (**upcoming**)

sneak peek
```java
public static void main(String[] args) throws IOException{
        Mab mab = new Mab();
        mab.run();
}
```
`
public static void main(String[] args) throws IOException{
        Mab mab = new Mab();
        mab.run();
}
`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

mogmyij committed:
added new file mab.java, removed Duke.java

mogmyij committed:
mab.java: wadded Echo functionality

mogmyij committed:
mab.java: added list functionality

mogmyij committed:
Added Task.java: task class for list \n cahnged name of mab.java to Mab.java to follow class name conventions

mogmyij committed:
Mab.java: changed to work with the parser and command class
added: Task subclasses (event, deadline, todo)
added: parser and command class

mogmyij committed:
added: testing script

mogmyij committed:
Command.java: added exception handling
added new file: MabException.java

mogmyij committed:
changed: Command.java to add delete capability

mogmyij committed:
add MabStorage.java: class used to manage the storage of data

mogmyij committed:
add new commands classes for each possible command
add load and store feature for tasks

mogmyij committed:
Merge branch 'branch-Level-7'

mogmyij committed:
fix DeleteCommand.java: command did not save changes after deletion

mogmyij committed:
add DateTimeUtil.java: utilities for parsing and handling date time

mogmyij committed:
Merge branch 'branch-Level-8'

mogmyij committed:
add: Ui.java, TaskList.java for better OOP

mogmyij committed:
moved all files into their relavant package directory

mogmyij committed:
Merge branch 'add-gradle-support'

mogmyij committed:
updated gradle files to match mab project

mogmyij committed:
add JUnit tests for the Deadlines task and DateTimeUtil class

mogmyij committed:
add javadocs to methods and classes

mogmyij committed:
Merge branch 'branch-A-JavaDoc'

mogmyij committed:
add FindCommand.java to repreesnt the find command
modified parser to recognise new find command.

mogmyij committed:
Merge branch 'branch-Level-9'

mogmyij committed:
change naming convention and package import order to follow coding standard

mogmyij committed:
Merge branch 'branch-A-CheckStyle'

mogmyij committed:
Add new package mab.gui to create GUI application for mab.

mogmyij committed:
Add FXML support for easier node styling.

WhyAre commented on src/main/java/mab/command/DeleteCommand.java:
```diff
@@ -0,0 +1,43 @@
+     */
+    @Override
+    public void execute(ArrayList<Task> list) throws MabException {
+        if (args.isBlank()) throw new MabException("argument cannot be empty");
```
```suggestion
        if (args.isBlank()) {
            throw new MabException("argument cannot be empty");
        }
```

Remember to put curly bracket even though there's only one statement

WhyAre commented on src/main/java/mab/task/Task.java:
```diff
@@ -0,0 +1,63 @@
+     * @param d Initial completion status
+     */
+    public Task(String t, boolean d) {
+        text = t; done = d;
```
Statements should be on separate lines
```suggestion
        text = t; 
        done = d;
```

mogmyij replied:
Thanks for the suggestions!

nguyenvukhang commented on text-ui-test/done.txt:
```diff
@@ -0,0 +1,92 @@
+██╗  ██╗███████╗██╗     ██╗      ██████╗  ██╗        ██╗███╗   ███╗    ███╗   ███╗ █████╗ ██████╗
+██║  ██║██╔════╝██║     ██║     ██╔═══██╗ ██║        ██║████╗ ████║    ████╗ ████║██╔══██╗██╔══██╗
+███████║█████╗  ██║     ██║     ██║   ██║ ██║        ██║██╔████╔██║    ██╔████╔██║███████║██████╔╝
+██╔══██║██╔══╝  ██║     ██║     ██║   ██║ ╚═╝        ██║██║╚██╔╝██║    ██║╚██╔╝██║██╔══██║██╔══██╗
+██║  ██║███████╗███████╗███████╗╚██████╔╝ ██╗        ██║██║ ╚═╝ ██║    ██║ ╚═╝ ██║██║  ██║██████╔╝
+╚═╝  ╚═╝╚══════╝╚══════╝╚══════╝ ╚═════╝  ╚═╝        ╚═╝╚═╝     ╚═╝    ╚═╝     ╚═╝╚═╝  ╚═╝╚═════╝
```
Consider making tests concise to reduce error logs when debugging.

mogmyij committed:
Merge branch 'branch-Level-10'

mogmyij committed:
add assertions to ensure correctness

mogmyij committed:
Merge pull request #1 from mogmyij/branch-Assertions

add assertions to ensure correctness, LGTM

mogmyij committed:
Code does is not up to code quality standard.

This could impact maintanability and readability.

Better comments that describe what and why have been added. better seperation is used to improve logic flow

It allows readers to have an easier time to understand the code

mogmyij committed:
Merge pull request #2 from mogmyij/branch-A-CodeQuality

Code does is not up to code quality standard. LTGM

mogmyij committed:
add sort command that sorts task list based on the start date time of the task

mogmyij committed:
Merge branch 'branch-BCD-Extension'

mogmyij committed:
Changed command execution to return strings

So that the GUI can make use of the strings.

Since the old method was printing to the CLI.

mogmyij committed:
Add better GUI design

Add CSS support for easier to use design.

In addition to FMXL makes it easier to modify the UI.

mogmyij committed:
Merge branch 'branch-A-BetterGui'

mogmyij committed:
Add personalised phrases

mogmyij committed:
Merge branch 'master' into branch-A-MoreErrorHandling

mogmyij committed:
Add more error handling

Detect the corruption of save file, and invalid events.

Ensures save file follows the correct format. And start time cannot be after end time.

mogmyij committed:
Merge branch 'branch-A-MoreErrorHandling'

mogmyij committed:
Add JUnit tests for commands

Test code was added for all commands.

Prevents regression of code.

mogmyij committed:
Merge branch 'branch-A-MoreTesting'

mogmyij committed:
Update README.md

Change the old duke readme to reflect the new Mab app.

More informative for users who are using Mab not Duke.

mogmyij committed:
change /doc to include more info

mogmyij committed:
change /doc to fix formattingw

mogmyij committed:
Change docs file for clarity

mogmyij committed:
change /doc to fix emoji
